src/backends/multicore/stimulus.cpp

```diff
@@ -23,19 +21,13 @@ class stimulus: public arb::multicore::mechanism {
     void nrn_state() override {}
     void nrn_current() override {
         size_type n = size();
-        // TODO: hackity
-        auto vec_t_range = util::make_range(vec_t_, vec_t_+n);
-        auto vec_i_range = util::make_range(vec_i_, vec_t_+n);
-        auto vec_ci_range = util::make_range(vec_ci_, vec_t_+n);
-
-        auto vec_t = util::indirect_view(util::indirect_view(vec_t_range, vec_ci_range), node_index_);
-        auto vec_i = util::indirect_view(vec_i_range, node_index_);
-
         for (size_type i=0; i<n; ++i) {
-            auto t = vec_t[i];
+            auto cv = node_index_[i];
+            auto t = vec_t_[vec_ci_[cv]];
```

Comment:
Good catch.

Reply:
It was all okay, I think, but the underscore vs no-underscore indirect view wrangling was confusing and unnecessary — it was put in as a quick way to get things working with the pointer representations originally.